features/step_definitions/view_restaurant_page_steps.rb

```diff
@@ -1,6 +1,9 @@
 Given("the following restaurants exist") do |table|
   table.hashes.each do |hash|
-    FactoryGirl.create(:restaurant, hash)
+    res_category = RestaurantCategory.find { |category| category.name == hash[:restaurant_category]}
+    hash.delete_if {|key, value| value == res_category.name }
+    restaurant = FactoryGirl.create(:restaurant, hash)
+    res_category.update(restaurants: [restaurant])
```

Comment:
This is not bad. I would refactor it though
```ruby
res_category = RestaurantCategory.find_by(name: hash[:restaurant_category])
hash.delete(:restaurant_category)
restaurant = FactoryGirl.create(:restaurant, hash)
restaurant.uppdate_attribute(:restaurant_category, res_category)
```
(Not tested in console. Wrote this from memory...)

Reply:
It all worked fine. Except the hash.delete_if part 
Tried a lot of methods to do this. for some reason they fail with the hash variable. If I create another hash in pry and use the methods like except and delete_if they work as intended.